### PR TITLE
Fix snipping nan filling for multiprocessing

### PR DESCRIPTION
**Pull request by Phlya**
Addressing https://github.com/open2c/cooltools/issues/335, for now in a straightforward way of storing locations of bad bins in a dict per region

### CODE REVIEW

File: cooltools/api/snipping.py [hunk 1]
```diff
@@ -305,10 +300,7 @@ def __init__(self, clr, cooler_opts=None, view_df=None, min_diag=2):
             # Make sure view_df is a proper viewframe
             try:
                 _ = is_compatible_viewframe(
-                    view_df,
-                    clr,
-                    check_sorting=True,
-                    raise_errors=True,
+                    view_df, clr, check_sorting=True, raise_errors=True,
```
Comment (sergpolly) [inline cooltools/api/snipping.py:303]: thank you , black !

File: cooltools/api/snipping.py [hunk 2]
```diff
@@ -351,19 +344,23 @@ def select(self, region1, region2):
             region1_coords, region2_coords
```
Comment (sergpolly) [inline cooltools/api/snipping.py:344]: could you please please leave a comment here that matrix is fetched sparse here ?

**Comment by sergpolly** [comment]
actually we have to double check if this is an issue at all ...
depending on how multiprocessing is used - it might actually create as many copies of snipper-instance as there are processes and nothing would overwrite the parent-objewct-instance ...

we just have to double check the initial premise of this 

one argument against is that if different chile-processes were indeed overwriting the parent `self._isnan` atribute - that something should have broke - just because regions have different length

**Comment by Phlya** [comment]
See [discussion](https://github.com/open2c/cooltools/discussions/379)

**Pull request closed without merging**